### PR TITLE
Add max_multipart_parts config option

### LINKED ISSUE #61: Configuring different max part numbers for some S3 providers
When setting up zfs_uploader against [Scaleway Object Storage](https://www.scaleway.com/en/object-storage/) (Specifically their `GLACIER` tier), everything worked as expected except with one caveat: The max part number on Scaleway is 1000 rather than the 10000 used by AWS.

This resulted in an error when uploading with the default setup, since it calculated the part sizes based on 10,000 parts and eventually failed due to exceeding Scaleway's limit of 1,000 parts. 

I resolved this for my use-case by simply modifying a number in job.py: https://github.com/Erisa/zfs_uploader/commit/20ed42fa5d64523484c58e62d6f3551dee6c40d2 however I feel that going forward it would be a good idea to allow configuration of this value in the zfs_uploader configuration file, and document it on the README.

You could also detect and change the values based on predefined provider limits, however it still would be nice to have the value in a user-configurable place.

### PR DESCRIPTION
**Pull request by Erisa**
Resolves #61 by implementing a `max_multipart_parts` configuration option, allowing users to customise this value to better suit other S3 providers with different max part sizes.

This PR also:
- Adds a Scaleway example to the README, using the new parts option.
- Adds `.venv/*` to gitignore because my setup decided to create it and it became a headache with git. Since this is the standard directory for venvs, it probably is a good idea to ignore it.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -1,2 +1,3 @@
 config.cfg
-.idea/*
\ No newline at end of file
+.idea/*
+.venv/*
```
Comment (ddebeau) [inline .gitignore:3]: Looks good but could use a newline at the end.

File: README.md [hunk 2]
```diff
@@ -119,6 +121,24 @@ max_incremental_backups_per_full = 6
 max_backups = 7
 ```
 
+#### Scaleway S3-compatible Object Storage, full backups
```
Comment (ddebeau) [inline README.md:124]: I'd like to keep the readme as storage provider agnostic as possible. I think it's fine to just document the `max_multipart_parts` parameter.

**Review by ddebeau — CHANGES_REQUESTED** [review]
I have a couple change requests but otherwise the PR looks good!

File: .gitignore (hunk 1 above)
Comment (Erisa) [reply .gitignore:3]: Sure thing, there wasn't one originally which was why I didn't pay much attention to there not being one on my commit either.

File: README.md (hunk 2 above)
Comment (Erisa) [reply README.md:124]: Absolutely fine, thanks for the feedback

**Comment by Erisa** [comment]
I also noticed the failing check which complains about some formatting (line too long, etc).

Is something you'd like me to resolve as well?

**Comment by ddebeau** [comment]
I can take care of those. Thanks for the PR!

**Pull request merged**